lib/matplotlib/artist.py

```diff
@@ -1161,12 +1135,57 @@ def properties(self):
         """Return a dictionary of all the properties of the artist."""
         return ArtistInspector(self).properties()
 
+    def _update_props(self, props, errorfmt):
+        """
+        Helper for `set` and `update`.
+
+        *errorfmt* is used to generate error messages for invalid property
+        names; it get formatted with ``type(self)`` and the property name.
```

Comment:
I suggest to use keywords instead of positional arguments because definition and resolution are quite far apart. Naming the parameters helps to understand what is what and reduces the risk that somebody will mess this up later.

Reply:
sure

Reply:
Your comment sounds like you want to do this, but the code has not yet changed.

Reply:
It has? errfmt is now kwonly (I'd argue there's little risk of misunderstanding what props is given the context).  Or did I misunderstand your suggestion?

Reply:
Oh, sorry. I was thinking about the format string, but didn't communicate that propertly; i.e.
```
errfmt.format(class_=type(self), prop_name=k)
```
and
 ```
"{class_.__name__}.set() got an unexpected keyword argument {prop_name!r}"
```